### PR TITLE
[camera] flip/change camera while recording - platform interface

### PR DESCRIPTION
**Pull request by BradenBagby**
This is the split out PR for the platform interface changes for this PR: https://github.com/flutter/plugins/pull/6478

### CODE REVIEW

**Comment by BradenBagby** [comment]
@bparrishMines Wondering if you have a chance to look at this PR?

**Review by bparrishMines — CHANGES_REQUESTED** [review]
Everything looks good. Just a small nit and I can LGTM after

File: packages/camera/camera_platform_interface/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 2.4
```
Comment (bparrishMines) [inline packages/camera/camera_platform_interface/CHANGELOG.md:1]: Some tests are failing because you are using `2.4` and not `2.4.0`.

**Review by bparrishMines — CHANGES_REQUESTED** [review]

File: packages/camera/camera_platform_interface/CHANGELOG.md (hunk 1 above)
Comment (BradenBagby) [reply packages/camera/camera_platform_interface/CHANGELOG.md]: Yikes thanks I fixed

**Review by bparrishMines — APPROVED** [review]
LGTM

**Comment by bparrishMines** [comment]
Forgot about secondary reviewer. 

@camsim99 @stuartmorgan Could you do a secondary review?

**Review by camsim99 — APPROVED** [review]
LGTM!

**Pull request merged**